homeassistant/components/azure_devops/sensor.py

```diff
@@ -2,89 +2,227 @@
+        translation_key="source_version",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.source_version,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="status",
+        translation_key="status",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.status,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="queue_time",
+        translation_key="queue_time",
+        device_class=SensorDeviceClass.DATE,
+        entity_registry_enabled_default=False,
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: parse_datetime(build.queue_time),
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="start_time",
+        translation_key="start_time",
+        device_class=SensorDeviceClass.DATE,
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: parse_datetime(build.start_time),
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="finish_time",
+        translation_key="finish_time",
+        device_class=SensorDeviceClass.DATE,
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: parse_datetime(build.finish_time),
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="url",
+        translation_key="url",
+        attrs=None,
+        value=lambda build: build.links.web if build.links else None,
+    ),
+)
+
+
+def parse_datetime(value: str | None) -> datetime | None:
+    """Parse datetime string."""
+    if value is None:
+        return None
+
+    return dt_util.parse_datetime(value)
 
 
 async def async_setup_entry(
-    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Azure DevOps sensor based on a config entry."""
     coordinator, project = hass.data[DOMAIN][entry.entry_id]
+    initial_builds: list[DevOpsBuild] = coordinator.data
+
+    sensors: list[AzureDevOpsBuildSensor] = []
+
+    # Add build sensors
+    for key, build in enumerate(initial_builds):
+        if build.project is None or build.definition is None:
+            _LOGGER.warning(
+                "Skipping build %s as it is missing a project or definition: %s",
+                key,
+                build,
+            )
+            continue
+
+        build_sensor_key_base = (
+            f"{build.project.project_id}_{build.definition.build_id}"
+        )
 
-    sensors = [
-        AzureDevOpsSensor(
-            coordinator,
-            AzureDevOpsSensorEntityDescription(
-                key=f"{build.project.project_id}_{build.definition.build_id}_latest_build",
-                translation_key="latest_build",
+        descriptions: list[AzureDevOpsBuildSensorEntityDescription] = [
```

Comment:
We can just make build a parameter for the `AzureDevOpsBuildSensor` and just pass that in together with the description

Reply:
I was going to make this a parameter (using an enum) when I add more types other than build.

Would this make more sense to leave the parameter until then?